internal/cli/cli.go

```diff
@@ -233,9 +234,63 @@ func documentGet(serverURL string, apiKey string, id string, idIsShareID bool) e
 
 	b, err := json.MarshalIndent(doc, "", "  ")
 	if err != nil {
-		return fmt.Errorf("failed marshalling document with id '%s: %w", id, err)
+		return fmt.Errorf("failed marshalling document with id '%s': %w", id, err)
 	}
 	fmt.Println(string(b))
 
 	return nil
 }
+
+func documentUpdate() *cobra.Command {
+	var title, text string
+	var append, publish bool
+
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update an existing document",
+		Long:  "Update an existing document's title, text etc. properties",
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			id := args[0]
+			errBase := fmt.Sprintf("failed updating document with ID '%s'", id)
+
+			// Extract value of global flags
+			key, err := c.Flags().GetString(flagApiKey)
+			if err != nil {
+				return fmt.Errorf("%s: %w", errBase, err)
+			}
+			url, err := c.Flags().GetString(flagServerURL)
+			if err != nil {
+				return fmt.Errorf("%s: %w", errBase, err)
+			}
+
+			cl := outline.New(url, &http.Client{}, key).Documents().Update(outline.DocumentID(id))
+			cl.Append(append)
+			cl.Publish(publish)
+			cl.Text(text)
+			cl.Title(title)
```

Comment:
In case we omit values in the cli, will these be overriden?
Assuming I have already a document with a title and a text.
Now I wanted to update the title only.
Running with
```
go run ./cli --key XXXXX  --server https://docs.io.ki  document update --publish --title "New Title" [ID]
```
would it remove the text? 🤔 
As the default value is empty string (`""`)?

Reply:
The corresponding data structs have `omitempty`  in JSON tags hence it is safe to use default `""` value of title and text flags. Using values of other flags without any checks is also safe and natural to. For example not using `--publish` should obviously mean that user wants to un-publish the doc.

Unless we make cli more complicated by having 3 choices: `publish` or `unpublish` or `do nothing if no flag given`. In this new way we would have to first download the doc examine its properties and then choose what to change. As I said too complicated for a simple task.